lib/schema.js

```diff
@@ -8,23 +8,39 @@ const DEFAULT_TASK_TIMEOUT = 30000;
 const DEFAULT_PROCESS_TIMEOUT = 3000;
 const DEFAULT_PREFETCH = 100;
 
+const handlerSchema = Joi.func().required();
+
 const configurationSchema = Joi.object({
-  handler: Joi.func().required(),
+  handlers: Joi.array().items(Joi.object().keys({
+    handle: handlerSchema,
+    validate: Joi.func().required(),
+    routingKey: Joi.string().required()
+  })).required(),
   workerName: Joi.string().required(),
   amqpUrl: Joi.string().required(),
   exchangeName: Joi.string().required(),
-  queueName: Joi.string().required(),
-  routingKey: Joi.string().required()
+  queueName: Joi.string().required()
 });
 
 const optionsSchema = Joi.object({
-  validator: Joi.func().default(_.noop),
   heartbeat: Joi.number().positive().default(DEFAULT_HEARTBEAT),
   taskTimeout: Joi.number().positive().default(DEFAULT_TASK_TIMEOUT),
   processExitTimeout: Joi.number().positive().default(DEFAULT_PROCESS_TIMEOUT),
   channelPrefetch: Joi.number().positive().default(DEFAULT_PREFETCH)
 });
 
+const legacyConfigurationSchema = Joi.object({
```

Comment:
the "legacy" prefix is less explicit than the "createWorker" prefix. Maybe put all the "createWorker" stuff in a file and all the "createWorkers" stuff in another file, so we can simpliy say "configurationSchema" everywhere ?